### DISCUSSION TITLE
How to allow privacy permissions in macOS Runner?

### DISCUSSION THREAD

**Discussion by natsuk4ze** [actions/virtual-environments#8961] [General questions]
Our macOS app test requires privacy permissions to the Photos app, is there any way to allow this in macOS Runner?

**Comment by ilia-shipitsin** [comment]
Hi @natsuk4ze!

You can learn more about TCC and service access rights [here](https://www.rainforestqa.com/blog/macos-tcc-db-deep-dive). Although it's not an official guide, it provides valuable insights for this purpose. Additionally, you can review how we apply permissions during the build [here](https://github.com/actions/runner-images/blob/main/images/macos/scripts/build/configure-tccdb-macos.sh).

We recommend checking exactly which permissions are missing. Together, we can then discuss potential solutions. You can perform diagnostics in two ways:

- Repeat the necessary steps on a local device running macOS and check the contents of TCC.db after a successful result.
- Obtain the necessary logs with the runner's denial.

The second method is somewhat more complex, so I'll describe it in a bit more detail:

- Execute your test/action that lacks Photos app access permissions to trigger the TCC prompt.
- Collect details by running sudo log show --last 15m | grep 'tccd: [com.apple.TCC:access] Prompting policy for hardened runtime'.

After it you should add the required permissions based on the received data. Resolving this may not be straightforward because at the moment, as of macOS 13, only the user's TCC database is available for modification.

**Reply by natsuk4ze** [reply]
Thanks for your kind support 😊 I will check it out.
